packages/core/src/shared/fs/fs.ts

```diff
@@ -182,24 +185,32 @@ export class FileSystem {
      *
      * @param path File location
      * @param data File content
-     * @param opt File permissions/flags. Only works in a non-web (nodejs) context. If provided,
-     * nodejs filesystem interface is used instead of routing through vscode VFS.
+     * @param opts.atomic If true, ensures content is not corrupted from a concurrent write.
+     *                    This is not truly atomic as an async write can override the final result,
+     *                    but if the content is structured like JSON it will still be parseable.
```

Comment:
can this be clarified? why would it matter what the contents (JSON or otherwise) are? either it can be overwritten or not. and if it's "atomic", then sure, the "peer" write will fully overwrite the file.

Reply:
The original reason for this "atomic" code was that our SSO cache file (JSON) was unparseable since it was malformed, eg there was an extra `}`. We guesses the case was that there were competing writes. We are fine if an overwrite happens, but wanted to avoid malformed content.

Does this clarify things?